bin/oneoff/import_concept_difficulty_tags.rb

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+require 'cdo/properties'
+require 'csv'
+
+# Import level concept difficulty tags from a gsheet (downloaded as a series of
+# CSVs). When we cloned the CSF courses from 2017 to 2018, the LCDs did not get
+# cloned with them so we're doing a mass import.
+
+# The given CSV contained headers that didn't map exactly to our defined LCDs,
+# so we perform a simple mapping in code
+HEADERS_TO_CONCEPTS = {
+  sequence_algorithms: ConceptDifficulties::SEQUENCING,
+  repeat: ConceptDifficulties::REPEAT_LOOPS,
+  functions_wparams: ConceptDifficulties::FUNCTIONS_WITH_PARAMS
+}
+
+def main(csv_dir)
+  %w(CourseA CourseB CourseC CourseD CourseE CourseF).each do |course|
+    CSV.foreach("#{csv_dir}/#{course}.csv", headers: true,  header_converters: :symbol) do |row|
+      concept_difficulties = Hash[row.to_hash.compact.map do |key, val|
+        [HEADERS_TO_CONCEPTS.fetch(key, key.to_s), val]
+      end].delete_if do |key, _val|
+        !LevelConceptDifficulty::CONCEPTS.include?(key)
+      end
+
+      level = Level.find_by(name: row.fetch(:level_name))
+      unless level
+        puts "cannot find #{row.fetch(:level_name)}"
+        next
+      end
+
+      lcd = LevelConceptDifficulty.find_or_create_by(level: level)
+      lcd.update!(concept_difficulties)
+      level.write_custom_level_file
```

Comment:
Only need to do this if we have a LCD for this level, right?  Seeing a lot of edits on levels that don't actually change the LCD.

Reply:
Hmm, good catch. Lemme try and make this a bit smarter

Reply:
Probably doesn't make a huge difference — the real (long-term) work item here is to not write values to the .level file unless they're different than the default value.

Reply:
Updated!